CodeJam.Main/ConnectionStrings/ConnectionStringBase.ConnectionStringBaseTypeConverter.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq.Expressions;
+
+using CodeJam.Reflection;
+
+namespace CodeJam.ConnectionStrings
+{
+	[TypeConverter(typeof(ConnectionStringBaseTypeConverter))]
+	public abstract partial class ConnectionStringBase
+	{
+		/// <inheritdoc />
+		public class ConnectionStringBaseTypeConverter : TypeConverter
+		{
+			private static readonly ConcurrentDictionary<Type, Func<string, object>> _creators = new();
+
+			private readonly Type _callingType;
+
+			public ConnectionStringBaseTypeConverter(Type callingType)
+			{
+				Code.AssertArgument(
+					callingType.IsAssignableTo<ConnectionStringBase>(), nameof(callingType),
+					$"To be used with this converter, the type must be inherited from {nameof(ConnectionStringBase)}");
+				_callingType = callingType;
+			}
+
+			private Func<string, object> GetCreator() => _creators.GetOrAdd(
+				_callingType, ct =>
```

Comment:
I'm not sure but there may be some helpers for object creation in the CJ. InfoOf / TypeDescriptor if I rememper it right.

If there are nothig suitable, the code looks good for me^)

Reply:
If you mean TypeAccessor<T>.Create, unfortunately this method only supports constructors without parameters. You can add overloads Create<TParam1, TParam2, ..>, of course, but that's a separate PR.